### PR TITLE
[feat] Add new logging handler for sending test data in JSON to an HTTP server

### LINKED ISSUE #1962: Create a new logging handler which will send a json directly via HTTP
We would need to be able to send the performance log as a json, preferrably using `requests` to handle the underlying complecity. The important requirement is to be able to use ReFrame's `encode` from `reframe.utility.jsonext` for ReFrame specific objects. The adapter should be able to accept a host, a path (for the url) and a port and probably some extra data to be added to the json such as `timestamp`.

### PR DESCRIPTION
**Pull request by teojgo**
Fixes #1962

### CODE REVIEW

**Comment by teojgo** [comment]
I need also to update the documentation

File: reframe/frontend/cli.py [hunk 1]
```diff
@@ -475,6 +475,18 @@ def main():
...
+        envvar='RFM_HTTPJSON_ADDRESS',
+        configvar='logging/handlers_perflog/httpjson_address',
+        help='HTTP Json server address'
+    )
+    argparser.add_argument(
+        dest='httpjson_url',
+        envvar='RFM_HTTPJSON_URL',
+        configvar='logging/handlers_perflog/httpjson_url',
+        help='HTTP Json url'
+    )
```
Comment (vkarak) [inline reframe/frontend/cli.py:489]: I think we should a URI to encompass everything.

**Review by vkarak — CHANGES_REQUESTED** [review]

File: docs/config_reference.rst [hunk 2]
```diff
@@ -1000,6 +1002,55 @@ The additional properties for the ``syslog`` handler are the following:
...
+This handler sends log records in JSON format to a server using HTTP POST requests.
+The additional properties for the ``httpjson`` handler are the following:
+
+.. js:attribute:: .logging[].handlers[].uri
+
+.. object:: .logging[].handlers_perflog[].uri
+
+   :required: Yes
+
+   The uri to be used in the http(s) request server defined as ``http(s)://host:port/url``.
```
Comment (vkarak) [inline docs/config_reference.rst:1018]:
```suggestion
   The URI to be used in the HTTP(S) request server defined as ``schema://host:port/path``.
```

File: docs/config_reference.rst [hunk 3]
```diff
@@ -1000,6 +1002,55 @@ The additional properties for the ``syslog`` handler are the following:
...
+.. object:: .logging[].handlers_perflog[].extras
+
+   :required: No
+   :default: ``{}``
+
+   A set of optional key/value pairs to be passed with each log record to the server.
+   These may depend on the server configuration.
+
+
+This log handler uses internally `requests <https://pypi.org/project/requests/>`__.
```
Comment (vkarak) [inline docs/config_reference.rst:1032]: I don't think that this information is useful for the user documentation.

File: docs/config_reference.rst [hunk 4]
```diff
@@ -1000,6 +1002,55 @@ The additional properties for the ``syslog`` handler are the following:
...
+The ``httpjson`` handler sends log messages in JSON format using an HTTP POST request to the specified uri.
```
Comment (vkarak) [inline docs/config_reference.rst:1033]:
```suggestion
The ``httpjson`` handler sends log messages in JSON format using an HTTP POST request to the specified URI.
```

File: docs/config_reference.rst [hunk 5]
```diff
@@ -1000,6 +1002,55 @@ The additional properties for the ``syslog`` handler are the following:
...
+       'level': 'info',
+       'format': '%(message)s',
+       'extras': {
+           'facility': 'reframe',
+           'data-version': '1.0'
+       }
+   }
+
+
+Although the ``format`` is defined for this handler, it is not only the log message that will be transmitted.
```
Comment (vkarak) [inline docs/config_reference.rst:1051]: Do we need to set the `format`?

**Review by vkarak — CHANGES_REQUESTED** [review]

**Pull request merged**